File: algorithms/graphics/dda_line/golang/line/dda_line.go
package line

import "math"

func DDA_line(x1 float64, y1 float64, x2 float64, y2 float64) ([]float64, []float64) {
	var i int = 0
	var x []float64
	var y []float64
	var x_start = math.Round(x1)
	var y_start = math.Round(y1)
	var x_end = math.Round(x2)
	var y_end = math.Round(y2)
	var length = max(math.Abs(x_end-x_start), math.Abs((y_end - y_start)))

	var dX = (x2 - x1) / length
	var dY = (y2 - y1) / length

	x = append(x, x1)
	y = append(y, y1)
	i++

	for i < int(length) {
		x = append(x, x[i-1]+dX)
		y = append(y, y[i-1]+dY)
		i++
	}
	x = append(x, x2)
	y = append(y, y2)

	return x, y
}

File: algorithms/graphics/dda_line/javascript/graphics_agloritms/dda_line.js
function dda_line(x1, y1, x2, y2){
    let i = 0;
    let x = Array();
    let y = Array();

    let x_start = Math.round(x1);
    let y_start = Math.round(y1);
    let x_end = Math.round(x2);
    let y_end = Math.round(y2);

    let length = Math.max(Math.abs(x_end - x_start), Math.abs(y_end - y_start));

    let dX = (x2 - x1) / length;
    let dY = (y2 - y1) / length;

    x.push(x1);
    y.push(y1);
    i++;
    
    while(i < length){
        x.push(x[i - 1] + dX);
        y.push(y[i - 1] + dY);
        i++;
    }

    x.push(x2);
    y.push(y2);

    let result = Array();
    result.push(x,y);

    return result;
}

export  {dda_line};